// Download compiler from https://github.com/nanopb/nanopb/releases, choose nanopb-w.x.y.z-windows-x86.zip
// Extract the zip, then Run:
// cd C:\Users\User\Devel\MetalHead\firmware\interface-board-zephyr\application\src
// C:\Users\User\Downloads\nanopb-w.x.y.z-windows-x86\generator-bin\nanopb_generator.exe status_indicators_t.proto
// Move StatusIndicators.pb.c to src/ directory
syntax = "proto3";

message status_indicators_t {
  bool shear_cut = 1;
  bool shear_home = 2;
  bool tool_out = 3;
  bool tool_in = 4;
  bool feed_forward = 5;
  bool feed_reverse = 6;
  bool head_cw = 7;
  bool head_ccw = 8;
  bool head_out = 9;
  bool head_in = 10;

  bool encoder_feed_set = 11;
  bool encoder_feed_reset = 12;
  bool encoder_bend_set = 13;
  bool encoder_bend_reset = 14;
  bool proximity_head_out = 15;
  bool proximity_head_in = 16;
  bool proximity_shear_home = 17;
  bool proximity_shear_cut = 18;
  bool proximity_tool_in = 19;
  bool proximity_tool_out = 20;
  bool shear_button = 21;
  bool proximity_head_limit = 22;
  bool proximity_head_cw = 23;
  bool proximity_head_ccw = 24;
  bool auto_button = 25;
  bool e_stop = 26;
  bool fuse = 27;
  bool vraw24 = 28;
  bool aux24v = 29;

  int32 encoder_feed_count = 30;
  int32 encoder_bend_count = 31;

  float twenty_four_vraw_bus_voltage = 32;
  float twenty_four_vraw_bus_current = 33;
  float aux_twenty_four_v_bus_voltage = 34;
  float aux_twenty_four_v_bus_current = 35;
  float twelve_v_bus_voltage = 36;
  float twelve_v_bus_current = 37;
  float five_v_bus_voltage = 38;
  float five_v_bus_current = 39;
  float three_v_three_v_bus_voltage = 40;
  float three_v_three_v_bus_current = 41;
  float f_gnd_bus_voltage = 42;
  float f_gnd_bus_current = 43;
}